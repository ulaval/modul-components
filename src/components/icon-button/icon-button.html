<button class="m-icon-button"
        :class="{ 'm--is-skin-light': isSkinLight,
               'm--is-skin-link': isSkinLink,
               'm--is-skin-dark': isSkinDark,
               'm--is-skin-primary': isSkinPrimary,
               'm--is-skin-secondary': isSkinSecondary,
               'm--has-ripple': ripple,
               'm--is-disabled': disabled }"
        :disabled="disabled"
        type="button"
        @click="onClick"
        @focus="onFocus"
        @blur="onBlur"
        :title="title"
        :style="{ width: buttonSize, height: buttonSize, fontSize: iconSize }">
    <m-icon ref="icon" class="m-icon-button__icon" :name="iconName"></m-icon>
    <span class="m-icon-button__hidden" v-if="hasSlots"><slot></slot></span>
</button>
