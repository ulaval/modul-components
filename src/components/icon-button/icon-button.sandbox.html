<div>
    <h3>No context</h3>
    <p class="m-u--margin-bottom">
        Basic behavior
    </p>
    <p>
        <m-icon-button>
        </m-icon-button>
    </p>

    <hr />

    <h3>With context icon</h3>
    <p class="m-u--margin-bottom">
        Will display the icon passed
    </p>
    <p>
        <m-icon-button icon-name="m-svg__edit">
        </m-icon-button>
    </p>

    <hr />

    <h3>
        With button size
        &nbsp;<select v-model="buttonSize">
            <option value="44px"
                    :selected="'44px' === buttonSize">44px</option>
            <option value="50px"
                    :selected="'50px' === buttonSize">50px</option>
            <option value="60px"
                    :selected="'60px' === buttonSize">60px</option>
        </select>
    </h3>
    <p class="m-u--margin-bottom">
        Will adjust the button size in px
    </p>
    <p>
        <m-icon-button :button-size="buttonSize"
                       :skin="mIconButtonSkin.Primary">
        </m-icon-button>
    </p>

    <hr />

    <h3>
        With icon size
        &nbsp;<select v-model="iconSize">
            <option value="20px"
                    :selected="'20px' === iconSize">20px</option>
            <option value="30px"
                    :selected="'30px' === iconSize">30px</option>
            <option value="40px"
                    :selected="'60px' === iconSize">40px</option>
        </select>
    </h3>
    <p class="m-u--margin-bottom">
        Will adjust the icon size in px
    </p>
    <p>
        <m-icon-button :icon-size="iconSize"
                       :skin="mIconButtonSkin.Primary">
        </m-icon-button>
    </p>

    <hr />

    <h3>
        Available Skins
        &nbsp;<select v-model="iconButtonSkin">
            <option v-for="(element, i) in iconButtonSkinAsArray"
                    :value="element.value"
                    :selected="element.value === iconButtonSkin"
                    :key="i">{{element.value}}</option>
        </select>
    </h3>
    <p class="m-u--margin-bottom">
        Change the icon button skin
    </p>
    <p>
        <m-icon-button :skin="iconButtonSkin"
                       title="allo">
        </m-icon-button>
    </p>
</div>
