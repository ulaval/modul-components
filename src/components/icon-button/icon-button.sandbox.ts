import Vue, { PluginObject } from 'vue';
import { Component } from 'vue-property-decorator';

import { ICON_BUTTON_NAME } from '../component-names';
import WithRender from './icon-button.sandbox.html';

@WithRender
@Component
export class MIconButtonSandbox extends Vue {
}

const IconButtonSandboxPlugin: PluginObject<any> = {
    install(v, options): void {
        v.component(`${ICON_BUTTON_NAME}-sandbox`, MIconButtonSandbox);
    }
};

export default IconButtonSandboxPlugin;
