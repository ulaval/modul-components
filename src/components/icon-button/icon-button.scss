@import 'abstracts/variables';

.m-icon-button {
    position: relative;
    display: inline-flex;
    align-items: center;
    text-align: center;
    justify-content: center;
    vertical-align: middle;
    border-radius: 50%;

    &:active,
    &:focus {
        outline: none;
    }

    &.m--has-ripple {
        &::before {
            content: '';
        }
    }

    &:not(.m--is-disabled) {
        &.m--is-skin-light,
        &.m--is-skin-link,
        &.m--is-skin-dark {
            &::before {
                transition: width $m-transition-duration ease, height $m-transition-duration ease, opacity $m-transition-duration--s linear;
                transition-delay: 0.3s;
                transform: translate(-50%, -50%);
                position: absolute;
                top: 50%;
                left: 50%;
                z-index: 0;
                display: block;
                width: $m-touch-size / 2;
                height: $m-touch-size / 2;
                border-radius: 50%;
                opacity: 0;
            }

            &:active,
            &:focus {
                &::before {
                    opacity: 1;
                    transition-delay: 0s;
                    width: $m-touch-size;
                    height: $m-touch-size;
                }
            }
        }

        &.m--is-skin-dark {
            color: rgba($m-color--white, 0.6);

            &:hover {
                color: $m-color--white;
            }

            &::before {
                background: $m-color--grey-dark;
            }
        }

        &.m--is-skin-light {
            color: $m-color--grey-darker;

            &:hover {
                color: $m-color--text;
            }

            &::before {
                background: $m-color--interactive-lightest;
            }
        }

        &.m--is-skin-link {
            color: $m-color--interactive;
            stroke-width: 2;

            &:hover,
            &:focus {
                color: $m-color--interactive-light;
            }

            &:active {
                color: $m-color--interactive-dark;
            }

            &::before {
                background: $m-color--interactive-lightest;
            }
        }

        &.m--is-skin-primary,
        &.m--is-skin-secondary {
            @include m-button--hover();
        }

        &.m--is-skin-primary {
            @include m-button--primary();
        }

        &.m--is-skin-secondary {
            @include m-button--secondary();
        }
    }

    &.m--is-disabled {
        &.m--is-skin-light,
        &.m--is-skin-link,
        &.m--is-skin-dark {
            cursor: not-allowed;
            color: $m-color--disabled;
        }

        &.m--is-skin-primary,
        &.m--is-skin-secondary {
            @include m-button--disabled();
        }
    }

    &.m--is-skin-light,
    &.m--is-skin-link,
    &.m--is-skin-dark {
        transition: color $m-transition-duration ease;
        padding: 0;
        cursor: pointer;
        border: none;
        background: transparent;
    }

    &.m--is-skin-primary,
    &.m--is-skin-secondary {
        @include m-button();
    }

    &__icon {
        position: relative;

        @include m-is-ie() {
            position: absolute;
            top: 50%;
            left: 50%;
            transform: translate3d(-50%, -50%, 0);
        }
    }

    &__hidden {
        @include m-visually-hidden;
    }
}
