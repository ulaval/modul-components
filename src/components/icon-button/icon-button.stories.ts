import { withA11y } from '@storybook/addon-a11y';
import { boolean, select, text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/vue';
import Vue from 'vue';
import { componentsHierarchyRootSeparator } from '../../../conf/storybook/utils';
import { ICON_BUTTON_NAME } from '../component-names';
import IconButtonPlugin from './icon-button';

Vue.use(IconButtonPlugin);

declare module '@storybook/addon-knobs' {
    export function withKnobs(): any;
}

const ICON_BUTTON_SKINS: {} = {
    'light': 'light',
    'dark': 'dark',
    'primary': 'primary',
    'secondary': 'secondary',
    'link': 'link',
    'bold': 'bold'
};

const ICON_BUTTON_ICONNAMES: {} = {
    'm-svg__close-clear': 'm-svg__close-clear',
    'm-svg__profile': 'm-svg__profile',
    'm-svg__information': 'm-svg__information',
    'm-svg__confirmation': 'm-svg__confirmation',
    'm-svg__warning': 'm-svg__warning',
    'm-svg__error': 'm-svg__error',
    'm-svg__calendar': 'm-svg__calendar',
    'm-svg__hint': 'm-svg__hint',
    'm-svg__clock': 'm-svg__clock'
};

storiesOf(`${componentsHierarchyRootSeparator}${ICON_BUTTON_NAME}`, module)
    .addDecorator(withA11y)
    .addDecorator(withKnobs)
    .add('name (short)', () => ({
        template: '<m-icon-button></m-icon-button>'
    }))
    .add('title', () => ({
        props: {
            title: {
                default: text('title', 'This is a custom title')
            }
        },
        template: `<div>
                    <m-icon-button :title="title"></m-icon-button>
                    <br>
                    <span>Current title: <span style="color: red">{{ title }}</span></span>
                   </div>`
    }))
    .add('disabled', () => ({
        props: {
            disabled: {
                default: boolean('disabled', true)
            }
        },
        template: '<m-icon-button :disabled="disabled"></m-icon-button>'
    }))
    .add('buttonSize', () => ({
        props: {
            buttonSize: {
                default: text('size (px)', '44')
            }
        },
        template: '<m-icon-button :buttonSize="`${buttonSize}px`"></m-icon-button>'
    }))
    .add('iconSize', () => ({
        props: {
            iconSize: {
                default: text('size (px)', '20')
            }
        },
        template: '<m-icon-button :iconSize="`${iconSize}px`"></m-icon-button>'
    }))
    .add('iconName', () => ({
        props: {
            iconName: {
                default: select('icon name', ICON_BUTTON_ICONNAMES, 'm-svg__close-clear')
            }
        },
        template: '<m-icon-button :iconName="iconName"></m-icon-button>'
    }))
    .add('ripple', () => ({
        props: {
            ripple: {
                default: boolean('ripple', true)
            }
        },
        template: `<div>
                    <p>Ripple effect on click when :ripple=true</p>
                    <br>
                    <m-icon-button :title="title"></m-icon-button>
                   </div>`

    }));

storiesOf(`${componentsHierarchyRootSeparator}${ICON_BUTTON_NAME}/skin`, module)
    .addDecorator(withA11y)
    .addDecorator(withKnobs)
    .add('all skins (use knob)', () => ({
        props: {
            skin: {
                default: select('skin', ICON_BUTTON_SKINS, 'light')
            }
        },
        template: '<m-icon-button :skin="skin"></m-icon-button>'
    }))
    .add('skin="light"', () => ({
        template: '<m-icon-button skin="light"></m-icon-button>'
    }))
    .add('skin="dark"', () => ({
        template: '<m-icon-button skin="dark" style="background: black"></m-icon-button>'
    }))
    .add('skin="primary"', () => ({
        template: '<m-icon-button skin="primary"></m-icon-button>'
    }))
    .add('skin="secondary"', () => ({
        template: '<m-icon-button skin="secondary"></m-icon-button>'
    }))
    .add('skin="link"', () => ({
        template: '<m-icon-button skin="link"></m-icon-button>'
    }))
    .add('skin="bold"', () => ({
        template: '<m-icon-button skin="bold"></m-icon-button>'
    }));

storiesOf(`${componentsHierarchyRootSeparator}${ICON_BUTTON_NAME}/all props`, module)
    .addDecorator(withA11y)
    .addDecorator(withKnobs)
    .add('make your own', () => ({
        props: {
            skin: {
                default: select('skin', ICON_BUTTON_SKINS, 'light')
            },
            title: {
                default: text('title', 'This is some text')
            },
            disabled: {
                default: boolean('disabled', false)
            },
            buttonSize: {
                default: text('button size (px)', '44')
            },
            iconSize: {
                default: text('icon size (px)', '20')
            },
            iconName: {
                default: select('icon name', ICON_BUTTON_ICONNAMES, 'm-svg__close-clear')
            },
            ripple: {
                default: boolean('ripple', true)
            }
        },
        template: `<div>
                        <p><span style="color: blue">skin</span> is set to <span style="color: red">{{ skin }}</span></p>
                        <p><span style="color: blue">disabled</span> is set to <span style="color: red">{{ disabled }}</span></p>
                        <p><span style="color: blue">buttonSize</span> is set to <span style="color: red">{{ buttonSize }}</span></p>
                        <p><span style="color: blue">iconSize</span> is set to <span style="color: red">{{ iconSize }}</span></p>
                        <p><span style="color: blue">iconName</span> is set to <span style="color: red">{{ iconName }}</span></p>
                        <p><span style="color: blue">ripple</span> is set to <span style="color: red">{{ ripple }}</span></p>
                        <br>
                        <div style="width: 250px; border-bottom: 1px solid black"></div>
                        <br>
                        <m-icon-button :buttonSize="buttonSize" :disabled="disabled" :iconName="iconName" :iconSize="iconSize"
                        :title="title" :ripple="ripple" :skin="skin"></m-icon-button>
                  </div>`
    }));
