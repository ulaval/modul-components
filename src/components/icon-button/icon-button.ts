import Vue, { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';
import WithRender from './icon-button.html?style=./icon-button.scss';
import { ICON_BUTTON_NAME, ICON_NAME } from '../component-names';
import IconPlugin from '../icon/icon';

export enum MIconButtonSkin {
    Light = 'light',
    Dark = 'dark',
    Primary = 'primary',
    Secondary = 'secondary',
    Link = 'link'
}

@WithRender
@Component
export class MIconButton extends Vue {
    @Prop({
        default: MIconButtonSkin.Light,
        validator: value =>
            value == MIconButtonSkin.Light ||
            value == MIconButtonSkin.Dark ||
            value == MIconButtonSkin.Primary ||
            value == MIconButtonSkin.Secondary ||
            value == MIconButtonSkin.Link
    })
    public skin: MIconButtonSkin;
    @Prop()
    public disabled: boolean;
    @Prop({ default: '44px' })
    public buttonSize: string;
    @Prop({ default: 'default' })
    public iconName: string;
    @Prop({ default: '20px' })
    public iconSize: string;
    @Prop({ default: true })
    public ripple: boolean;
    @Prop()
    public title: string;

    protected mounted(): void {
        this.hasSlots();
    }

    private onClick(event: Event): void {
        this.$emit('click', event);
        this.$el.blur();
    }

    private onFocus(event: Event): void {
        this.$emit('focus');
    }

    private onBlur(event: Event): void {
        this.$emit('blur');
    }

    private get isSkinLight(): boolean {
        return this.skin == MIconButtonSkin.Light;
    }

    private get isSkinDark(): boolean {
        return this.skin == MIconButtonSkin.Dark;
    }

    private get isSkinPrimary(): boolean {
        return this.skin == MIconButtonSkin.Primary;
    }

    private get isSkinSecondary(): boolean {
        return this.skin == MIconButtonSkin.Secondary;
    }

    private get isSkinLink(): boolean {
        return this.skin == MIconButtonSkin.Link;
    }

    private hasSlots(): boolean {
        let hasSlot: boolean = !!this.$slots.default;
        if (!hasSlot) {
            // console.warn('<' + ICON_BUTTON_NAME + '> needs a text in its default slot that will describe its function. This text will be hidden and only read by the screen readers.');
        }
        return hasSlot;
    }
}

const IconButtonPlugin: PluginObject<any> = {
    install(v, options) {
        console.debug(ICON_BUTTON_NAME, 'plugin.install');
        console.warn(ICON_BUTTON_NAME + ' is not ready for production');
        v.use(IconPlugin);
        v.component(ICON_BUTTON_NAME, MIconButton);
    }
};

export default IconButtonPlugin;
