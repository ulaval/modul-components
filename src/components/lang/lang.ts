import Vue from 'vue';
import { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';
import WithRender from './lang.html';
import { I18N_NAME } from '../component-names';
import { ModulVue } from '../../utils/vue';

@WithRender
@Component
export class MLang extends ModulVue {
    public text: string = '';

    @Prop()
    public msg: string;

    public mounted() {
        this.text = this.$i18n.translate(this.msg);
    }
}

const LangPlugin: PluginObject<any> = {
    install(v, options) {
        v.component(I18N_NAME, MLang);
    }
};

export default LangPlugin;
