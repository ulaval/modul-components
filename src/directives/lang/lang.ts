import Vue, { VNodeDirective } from 'vue';
import { PluginObject } from 'vue';
import { I18N_NAME } from '../directives-names';
import Messages from '@/utils/i18n';

export class MI18n extends Vue {
    public bind(element: HTMLElement, binding: VNodeDirective) {
        element.innerText = Messages.translate(binding.value);
    }
}

const MI18nPlugin: PluginObject<any> = {
    install(v, options) {
        v.directive(I18N_NAME, new MI18n());
    }
};

export default MI18nPlugin;
