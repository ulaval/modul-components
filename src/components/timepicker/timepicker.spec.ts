import { createLocalVue, mount } from '@vue/test-utils';
import Vue, { VueConstructor } from 'vue';
import moment from 'moment';

import { resetModulPlugins } from '../../../tests/helpers/component';
import { addMessages } from '../../../tests/helpers/lang';
import { renderComponent } from '../../../tests/helpers/render';
import uuid from '../../utils/uuid/uuid';
import { MIconButton } from '../icon-button/icon-button';
import TimepickerPlugin, { MTimepicker } from './timepicker';

jest.mock('../../utils/uuid/uuid');
(uuid.generate as jest.Mock).mockReturnValue('uuid');

describe('MTimepicker', () => {
    let localVue: VueConstructor<Vue>;
    let mockDropDown: VueConstructor<Vue>;

    beforeEach(() => {
        resetModulPlugins();
        localVue = createLocalVue();
        Vue.use(TimepickerPlugin);
        addMessages(Vue, [
            'components/timepicker/timepicker.lang.en.json'
        ]);
    });

    it('should render correctly', () => {
        const tp = mount(MTimepicker, {
            localVue: localVue,
            propsData: {
                time: moment('02:00 PM', 'h:mm A')
            }
        });

        return expect(renderComponent(tp.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly when waiting is set', () => {
        const tp = mount(MTimepicker, {
            localVue: localVue,
            propsData: {
                waiting: true,
                time: moment('02:00 PM', 'h:mm A')
            }
        });

        return expect(renderComponent(tp.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly when disabled is set', () => {
        const tp = mount(MTimepicker, {
            localVue: localVue,
            propsData: {
                disabled: true,
                time: moment('02:00 PM', 'h:mm A')
            }
        });

        return expect(renderComponent(tp.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly when error is set', () => {
        const tp = mount(MTimepicker, {
            localVue: localVue,
            propsData: {
                error: true,
                time: moment('02:00 PM', 'h:mm A'),
                errorMessage: 'Nostrud laboris quis velit voluptate aute elit elit non.'
            }
        });

        return expect(renderComponent(tp.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly when valid is set', () => {
        const tp = mount(MTimepicker, {
            localVue: localVue,
            propsData: {
                valid: true,
                time: moment('02:00 PM', 'h:mm A'),
                validMessage: 'Nostrud laboris quis velit voluptate aute elit elit non.'
            }
        });

        return expect(renderComponent(tp.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly min/max time', () => {
        const tp = mount(MTimepicker, {
            localVue: localVue,
            propsData: {
                duration: true,
                time: moment('03:00', 'h:mm A'),
                min: moment.duration(2, 'hour'),
                max: moment.duration(7, 'hour')
            }
        });

        return expect(renderComponent(tp.vm)).resolves.toMatchSnapshot();
    });
});
