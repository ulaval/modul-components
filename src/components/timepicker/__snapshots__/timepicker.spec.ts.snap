// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MTimepicker should render correctly 1`] = `
<div class="m-timepicker">
    <div class="m-input-style m--has-value m--is-tag-default" style="width:undefined;">
        <div class="m-input-style__body">
            <div class="m-input-style__body__left">
                <input id="mTimepicker-uuid" type="text" aria-haspopup="true" aria-controls="mTimepicker-uuid-controls" placeholder="hh mm" value="2:00 PM"> </div>
            <div class="m-input-style__body__right">
                <m-icon-button icon-name="clock" button-size="22px" aria-controls="mTimepicker-uuid"><span class="m-i18n">Open time selector</span> </m-icon-button>
            </div>
        </div>
    </div>
    <div class="m-validation-message">
        <m-accordion-transition transition="true"></m-accordion-transition>
    </div>
    <div class="m-popup">
        <div class="m-popper">
            <DIV class="v-portal" style="display:none;"></DIV>
        </div>
    </div>
</div>
`;

exports[`MTimepicker should render correctly min/max time 1`] = `
<div class="m-timepicker">
    <div class="m-input-style m--has-value m--is-tag-default" style="width:undefined;">
        <div class="m-input-style__body">
            <div class="m-input-style__body__left">
                <input id="mTimepicker-uuid" type="text" aria-haspopup="true" aria-controls="mTimepicker-uuid-controls" placeholder="hh mm" value="7:00"> </div>
            <div class="m-input-style__body__right">
                <m-icon-button icon-name="clock" button-size="22px" aria-controls="mTimepicker-uuid"><span class="m-i18n">Open time selector</span> </m-icon-button>
            </div>
        </div>
    </div>
    <div class="m-validation-message">
        <m-accordion-transition transition="true"></m-accordion-transition>
    </div>
    <div class="m-popup">
        <div class="m-popper">
            <DIV class="v-portal" style="display:none;"></DIV>
        </div>
    </div>
</div>
`;

exports[`MTimepicker should render correctly when disabled is set 1`] = `
<div class="m-timepicker">
    <div class="m-input-style m--is-disabled m--has-value m--is-tag-default" style="width:undefined;">
        <div class="m-input-style__body">
            <div class="m-input-style__body__left">
                <input id="mTimepicker-uuid" type="text" aria-haspopup="true" aria-controls="mTimepicker-uuid-controls" disabled="disabled" placeholder="hh mm" value="2:00 PM"> </div>
            <div class="m-input-style__body__right">
                <m-icon-button icon-name="clock" disabled="disabled" button-size="22px" aria-controls="mTimepicker-uuid"><span class="m-i18n">Open time selector</span> </m-icon-button>
            </div>
            <svg aria-hidden="true" width="1em" height="1em" class="m-icon m-input-style__disabled-icon">
                <use xlink:href="#default" aria-hidden="true"></use>
            </svg>
        </div>
    </div>
    <div class="m-validation-message">
        <m-accordion-transition transition="true"></m-accordion-transition>
    </div>
    <div class="m-popup">
        <div class="m-popper">
            <DIV class="v-portal" style="display:none;"></DIV>
        </div>
    </div>
</div>
`;

exports[`MTimepicker should render correctly when error is set 1`] = `
<div class="m-timepicker">
    <div class="m-input-style m--has-error m--has-value m--is-tag-default" style="width:undefined;">
        <div class="m-input-style__body">
            <div class="m-input-style__body__left">
                <input id="mTimepicker-uuid" type="text" aria-haspopup="true" aria-controls="mTimepicker-uuid-controls" placeholder="hh mm" value="2:00 PM"> </div>
            <div class="m-input-style__body__right">
                <m-icon-button icon-name="clock" button-size="22px" aria-controls="mTimepicker-uuid"><span class="m-i18n">Open time selector</span> </m-icon-button>
            </div>
        </div>
    </div>
    <div class="m-validation-message">
        <m-accordion-transition transition="true">
            <div class="m-validation-message__wrap">
                <p class="m-validation-message__error">
                    <svg width="1em" height="1em" class="m-icon m-validation-message__error__icon">
                        <title>m-validation-message:title-error-icon</title>
                        <use xlink:href="#error" aria-hidden="true"></use>
                    </svg><span class="m-validation-message__text">Nostrud laboris quis velit voluptate aute elit elit non.</span></p>
            </div>
        </m-accordion-transition>
    </div>
    <div class="m-popup">
        <div class="m-popper">
            <DIV class="v-portal" style="display:none;"></DIV>
        </div>
    </div>
</div>
`;

exports[`MTimepicker should render correctly when valid is set 1`] = `
<div class="m-timepicker">
    <div class="m-input-style m--is-valid m--has-value m--is-tag-default" style="width:undefined;">
        <div class="m-input-style__body">
            <div class="m-input-style__body__left">
                <input id="mTimepicker-uuid" type="text" aria-haspopup="true" aria-controls="mTimepicker-uuid-controls" placeholder="hh mm" value="2:00 PM"> </div>
            <div class="m-input-style__body__right">
                <m-icon-button icon-name="clock" button-size="22px" aria-controls="mTimepicker-uuid"><span class="m-i18n">Open time selector</span> </m-icon-button>
            </div>
        </div>
    </div>
    <div class="m-validation-message">
        <m-accordion-transition transition="true">
            <div class="m-validation-message__wrap">
                <p class="m-validation-message__valid">
                    <svg width="1em" height="1em" class="m-icon m-validation-message__valid__icon">
                        <title>m-validation-message:title-valid-icon</title>
                        <use xlink:href="#check" aria-hidden="true"></use>
                    </svg> <span class="m-validation-message__text">Nostrud laboris quis velit voluptate aute elit elit non.</span></p>
            </div>
        </m-accordion-transition>
    </div>
    <div class="m-popup">
        <div class="m-popper">
            <DIV class="v-portal" style="display:none;"></DIV>
        </div>
    </div>
</div>
`;

exports[`MTimepicker should render correctly when waiting is set 1`] = `
<div class="m-timepicker">
    <div class="m-input-style m--is-waiting m--has-value m--is-tag-default" style="width:undefined;">
        <div class="m-input-style__body">
            <div class="m-input-style__body__left">
                <input id="mTimepicker-uuid" type="text" aria-haspopup="true" aria-controls="mTimepicker-uuid-controls" disabled="disabled" placeholder="hh mm" value="2:00 PM"> </div>
            <div class="m-input-style__body__right">
                <m-icon-button icon-name="clock" disabled="disabled" button-size="22px" aria-controls="mTimepicker-uuid"><span class="m-i18n">Open time selector</span> </m-icon-button>
            </div>
            <div class="m-spinner m-input-style__spinner">
                <div></div>
            </div>
        </div>
    </div>
    <div class="m-validation-message">
        <m-accordion-transition transition="true"></m-accordion-transition>
    </div>
    <div class="m-popup">
        <div class="m-popper">
            <DIV class="v-portal" style="display:none;"></DIV>
        </div>
    </div>
</div>
`;
